### DISCUSSION TITLE
Amazon Listing - Required Attributes Schema

### DISCUSSION THREAD

**Discussion by jparker1986** [amzn/selling-partner-api-models#4211] [General]
Hello,

I have been reading and testing the Amazon SP API and wondered if anyone could clarify how to make listing a product on Amazon simpler using the API.

I am fetching the node and Schema for that category by using the GET {{baseUrl}}/definitions/2020-09-01/productTypes/OUTDOOR_LIVING?marketplaceIds=XXXXX

The API returns the schema with all the attributes but does not specify which ones are mandatory. 

The only way I can find a workaround is by listing the product using VALIDATION PREVIEW mode and letting Amazon tell me which attributes are required. 

Is there any method to actually see which attributes are mandatory before listing the product?

**Comment by weilinggu** [comment]
@jparker1986 The required attributes are dynamic. As you fill out different attributes, the required attributes might change. A few resources that could help. 
- [Here](https://youtu.be/fraQT3UhFII?si=K4tCPUOA3gILc7mU) is a deep dive video on product type JSON schema that explains how it works
- [Here](https://github.com/amzn/selling-partner-api-samples/tree/main/use-cases/listings) is a Listings sample application that implemented a dynamic UI based on product type JSON schema. 

[json-schema.org](https://json-schema.org/tools?query=&sortBy=name&sortOrder=ascending&groupBy=toolingTypes&licenses=&languages=&drafts=&toolingTypes=&environments=) has tools you can use to render the product type JSON schema and validate your product payload. The recommendation is to validate your product payload using these tools before calling VALIDATION PREVIEW.

Hope this helps!

Weiling
Selling Partner Developer Services
